WordPress/Classes/ViewRelated/Comments/CommentDetailInfoViewController.swift

```diff
@@ -100,24 +104,23 @@ extension CommentDetailInfoViewController: UITableViewDelegate {
     }
 }
 
-// MARK: - DrawerPresentable
-extension CommentDetailInfoViewController: DrawerPresentable {
-    var collapsedHeight: DrawerHeight {
-        .intrinsicHeight
-    }
-
-    var allowsUserTransition: Bool {
-        false
-    }
-
-    var compactWidth: DrawerWidth {
-        .maxWidth
-    }
-}
-
-// MARK: - ChildDrawerPositionable
-extension CommentDetailInfoViewController: ChildDrawerPositionable {
-    var preferredDrawerPosition: DrawerPosition {
-        .collapsed
+extension CommentDetailInfoViewController {
+    func show(from presentingViewController: UIViewController, sourceView: UIView) {
+        let navigationController = UINavigationController(rootViewController: self)
+        if UIDevice.isPad() {
+            navigationController.modalPresentationStyle = .popover
+            navigationController.popoverPresentationController?.sourceView = sourceView
+            navigationController.popoverPresentationController?.permittedArrowDirections = [.up]
+        } else {
+            self.navigationItem.leftBarButtonItem = UIBarButtonItem(title: SharedStrings.Button.close, primaryAction: UIAction { [weak presentingViewController] _ in
+                presentingViewController?.dismiss(animated: true)
+            })
+            if let sheet = navigationController.sheetPresentationController {
+                sheet.detents = [.medium(), .large()]
+                sheet.prefersGrabberVisible = true
+                navigationController.additionalSafeAreaInsets = UIEdgeInsets(top: 8, left: 0, bottom: 0, right: 0) // For grabber
+            }
```

Comment:
Instead of checking iPad or size class, maybe we can present the info either in sheet if available, and fallback to popover if sheet is not available?